service/src/main/java/com/kodemon/service/implementations/TrainerServiceImpl.java

```diff
@@ -79,6 +79,20 @@ public void delete(Long id) {
     }
 
     @Override
+    public void blockTrainer(Long id) {
```

Comment:
You could simplify this to just one method, eg.
`public void setBlocked(Long id, boolean blocked) {}`
as those methods consist of practically duplicate code, but I leave it up to you.